nmigen/build/run.py

```diff
@@ -99,6 +100,64 @@ def execute_local(self, root="build", *, run_script=True):
         finally:
             os.chdir(cwd)
 
+    def execute_remote_ssh(self, root, hostname, *, connect_args = {}):
```

Comment:
I think a signature like the following:
```suggestion
    def execute_remote_ssh(self, *, connect_to={}, root):
```
would work better: it doesn't require placing the directory before the hostname, and it lets you pass only a single dict to `RemoteSSHBuildProducts`.